port/common/omrsysinfo.c

```diff
@@ -163,6 +163,40 @@ omrsysinfo_get_processor_feature_name(struct OMRPortLibrary *portLibrary, uint32
 	return rc;
 }
 
+/**
+ * Generate the corresponding string literals for the provided OMRProcessorDesc.
+ *
+ * @param[in] portLibrary The port library.
+ * @param[out] buff The cpu features output string.
+ * @param[in] BUFF_SIZE The size of the buff.
+ */
+void
+omrsysinfo_print_processor_features(struct OMRPortLibrary *portLibrary, OMRProcessorDesc *desc, char * buff, const size_t BUFF_SIZE)
+{
+	memset(buff, 0, BUFF_SIZE*sizeof(char));
+	size_t rowLength = 0;
+	for (size_t i = 0; i < OMRPORT_SYSINFO_FEATURES_SIZE; i++) {
+		size_t numberOfBits = CHAR_BIT * sizeof(desc->features[i]);
+		for (size_t j = 0; j < numberOfBits; j++) {
+
+			if (desc->features[i] & (1<<j)) {
+
+				uint32_t feature = (uint32_t)(i * numberOfBits + j);
+				const char * feature_name = omrsysinfo_get_processor_feature_name(portLibrary, feature);
+				
+				if (rowLength > 0) {
+					strncat(buff, " ", BUFF_SIZE - strlen(buff) - 1);
+					rowLength += 1;
+				}
+				strncat(buff, feature_name, BUFF_SIZE - strlen(buff) - 1);
+				rowLength += strlen(feature_name);
+		    }
+		}
+	}
+	strncat(buff, "\n", BUFF_SIZE - strlen(buff) - 1);
+	return;
```

Comment:
The `return;` statement seems redundant as there are other examples of `void` functions in this file not using it.